python/pyxet/src/lib.rs

```diff
@@ -444,19 +466,30 @@ impl PyRFile {
     // why does IOBase have readline? this is not very nice.
     #[pyo3(signature = (size=-1))]
     pub fn readline(&mut self, size: i64, py: Python<'_>) -> PyResult<PyObject> {
-        let ret = self.readline_impl(size)?;
+        let ret = rust_async!(py, self.readline_impl(size).await)?;
         Ok(PyBytes::new(py, &ret).into())
     }
+
     #[pyo3(signature = (num_lines=-1))]
     pub fn readlines(&mut self, num_lines: i64, py: Python<'_>) -> PyResult<PyObject> {
-        let ret = PyList::empty(py);
-        while num_lines <= 0 || (ret.len() as i64) < num_lines {
-            let buf = self.readline_impl(-1)?;
-            if !buf.is_empty() {
-                let _ = ret.append(PyBytes::new(py, &buf));
-            } else {
-                break;
+        let v_buf = rust_async!(py, {
+            let mut v_buf = Vec::new();
+
+            while num_lines <= 0 || (v_buf.len() as i64) < num_lines {
+                let buf = self.readline_impl(-1).await?; // TODO: This has to be wrong re num_lines, given the API??
```

Comment:
readlines(n) does not do what one thinks it should do. Rather, `num_lines` is the wrong name for the argument. it should be "hint".
https://stackoverflow.com/questions/14541010/pythons-function-readlinesn-behavior

Reply:
yup. was just responding to the TODO comment

Reply:
Ah.  Gotcha.